tunnel-server/src/app.ts

```diff
@@ -3,20 +3,29 @@ import { fastifyRequestContext } from '@fastify/request-context'
 import http from 'http'
 import internal from 'stream'
 import {Logger} from 'pino'
+import { SessionManager } from './seesion'
+import { Claims, JwtAuthenticator, authenticator, unauthorized } from './auth'
+import { PreviewEnvStore } from './preview-env'
+import { match } from 'ts-pattern'
+import { replaceHostname } from './url'
+import { resolve } from 'url'
 
-export const app = ({ isProxyRequest, proxyHandlers, logger }: {
+export const app = ({ isProxyRequest, proxyHandlers,sessionManager, baseUrl, envStore, logger }: {
   isProxyRequest: (req: http.IncomingMessage) => boolean
   logger: Logger
+  baseUrl: URL
+  sessionManager: SessionManager<Claims>
+  envStore: PreviewEnvStore
   proxyHandlers: { wsHandler: (req: http.IncomingMessage, socket: internal.Duplex, head: Buffer) => void, handler: (req: http.IncomingMessage, res: http.ServerResponse) => void }
 }) =>
   Fastify({
     serverFactory: (handler) => {
       const {wsHandler:proxyWsHandler, handler: proxyHandler } = proxyHandlers
       const server = http.createServer((req, res) => {
-        if (isProxyRequest(req)){
-          return proxyHandler(req, res)
-        }
-        return handler(req, res)
+        return match(req)
```

Comment:
Not sure using a custom DSL like `match` is better than plain `if`s, matter of style I guess.

Also - the first `match` call can be omitted (since it's covered by the `otherwise` call)